cmd/ls.go

```diff
@@ -29,33 +28,29 @@ import (
 )
 
 // NewLsCmd returns a new ls command.
-func NewLsCmd(reader ConfigReader, ioStreams IOStreams) *cobra.Command {
+func NewLsCmd(targetReader TargetReader, configReader ConfigReader, ioStreams IOStreams) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "ls [gardens|projects|seeds|shoots|issues]",
-		Short: "List all resource instances, e.g. list of shoots|issues",
-		Run: func(cmd *cobra.Command, args []string) {
+		Use:          "ls [gardens|projects|seeds|shoots|issues]",
+		Short:        "List all resource instances, e.g. list of shoots|issues",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) < 1 || len(args) > 2 {
-				fmt.Println("Command must be in the format: ls [gardens|projects|seeds|shoots|issues]")
-				os.Exit(2)
+				return errors.New("command must be in the format: ls [gardens|projects|seeds|shoots|issues]")
```

Comment:
We could put those strings in a constant and then use it also for the expected results in the tests.

Reply:
I checked in `pkg/kubectl` of k/kubernetes and the convention there is to duplicate the literals for the test and non-tests.